codegen/gateway.go

```diff
@@ -420,7 +420,71 @@ func NewEndpointSpec(
 		ClientMethod:       clientMethod,
 	}
 
-	return augmentEndpointSpec(espec, endpointConfigObj, midSpecs)
+	mandatoryMidSpecs, err := getOrderedMandatoryMiddlewareSpecs(
+		h.ConfigRoot(),
+		h.MandatoryMiddlewareSpecs(),
+		endpointType.(string))
+	if err != nil {
+		return nil, errors.Wrap(
+			err, "Error getting ordered mandatory middleware specs",
```

Comment:
Error message usually should not start with capital letter because it's often wrapped by another error or prefixed by another message. https://github.com/golang/go/wiki/Errors#errors. (sadly the existing error strings in the repo are not the best examples)